pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -220,6 +219,14 @@ func MakePod(build *v1alpha1.Build, kubeclient kubernetes.Interface) (*corev1.Po
 	}
 	annotations["sidecar.istio.io/inject"] = "false"
 
+	labels := map[string]string{}
+	for key, val := range build.ObjectMeta.Labels {
+		labels[key] = val
+	}
+	// TODO: Redundant with TaskRun label set in `taskrun.makeLabels`. Should
```

Comment:
Previously `makelabels` function added an additional label to existing taskrun labels but the additional label is added in this line. So can we delete `makelabels` function and pass taskrun labels directly to build? does that make sense @dwnusbaum ?

Reply:
`makeLabels` and `MakePod` add a label with different keys, but the same value. The key added in `makeLabels` is `pipeline.knative.dev/taskRun`, and the one added in `MakePod` is `build.knative.dev/buildName`. I wasn't sure if we needed to keep the `buildName` label for compatibility or anything, but if not then yeah I think we should just delete it.

I think we need to keep `makeLabels`, since that passes labels from `TaskRun` to `Build`, so that `MakePod` can pass them from `Build` to `Pod`, but if we delete the extra label added in `MakePod` then we could just set `Pod.ObjectMeta.Labels = Build.ObjectMeta.Labels` so we don't need to create a new map in `MakePod`. Otherwise we could make `MakePod` take `TaskRun` instead of `Build` so that we could delete `makeLabels`, but that seems like more work. What do you think?

Reply:
Following up with this is #494.

Reply:
> I wasn't sure if we needed to keep the `buildName` label for compatibility or anything, but if not then yeah I think we should just delete it.

I think there is no value in adding buildName label anymore.

> I think we need to keep `makeLabels`,

I like this idea

> then we could just set `Pod.ObjectMeta.Labels = Build.ObjectMeta.Labels`

👍